### PR TITLE
Clojure migration to split v1 data perm paths into v2 data & query perms

### PR DESCRIPTION
**Pull request by noahmoss**
This is a migration to convert v1 data permission paths that existed in an instance's database prior to v46 into separate v2 data and query permission paths. It uses the new mechanism to run Clojure migrations via Liquibase, since `REGEXP_SUBSTR` isn't available on MySQL 5.7. 

I'm not using the existing permission path conversion logic in `metabase.models.permissions` because we're trying to keep these migrations from having minimal dependencies on other parts of the codebase, to keep them robust. Instead I've replicated that logic using a single raw regex for the most part. Essentially, I have a regex to parse almost any v1 data perms path and extract the "base path", which is the path for the resource without any suffixes. Then the corresponding v2 paths are just the base path with `/data` or `/query` prefixes. The one exception is `/db/:id/schema` since the v2 query path also needs the `/schema` suffix, so I have special logic for handling this.

I've also added a test to make sure that each path type is converted appropriately.

### CODE REVIEW

**Review by escherize — APPROVED** [review]

File: src/metabase/db/custom_migrations.clj [hunk 1]
```diff
@@ -52,3 +58,49 @@
...
+    ;; For (almost) all v1 data paths, we simply extract the base path (e.g. "/db/1/schema/PUBLIC/table/1/")
+    ;; and construct new v2 paths by adding prefixes to the base path.
+    [(str "/data" base-path) (str "/query" base-path)]
+
+    ;; For the specific v1 path that grants full data access but no native query access, we add a
+    ;; /schema/ suffix to the corresponding v2 query permission path.
+    (when-let [db-id (second (re-find #"^/db/(\d+)/schema/$" v1-path))]
+      [(str "/data/db/" db-id "/") (str "/query/db/" db-id "/schema/")])))
+
+(def-reversible-migration SplitDataPermissions
```
Comment (camsaul) [inline src/metabase/db/custom_migrations.clj:84]: Not really part of this PR but traditionally `def` macros that consist of more than one word spell out the whole word `define`, so this should have been called `define-reversible-migration`

Comment (noahmoss) [reply src/metabase/db/custom_migrations.clj:84]: 👍 I can change it in this PR
